src/main/java/com/example/demo/exercise/controller/ProductController.java

```diff
@@ -0,0 +1,70 @@
+package com.example.demo.exercise.controller;
+
+import com.example.demo.exercise.model.Category;
+import com.example.demo.exercise.model.Product;
+import com.example.demo.exercise.repository.ProductRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller
+public class ProductController {
+
+    private ProductRepository productRepository;
+
+    @Autowired
+    public ProductController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @ResponseBody
+    @GetMapping("/all-products")
+    public String getAll(){
+        List<Product> productList = productRepository.findAll();
+
+        String productsDetails =  productList.stream()
+                .map(Product::toString)
+                .collect(Collectors.joining("<br/>"));
+
+        String sumPrices = sumPrice(productList);
+        return productsDetails + sumPrices
+                + "<a href=\"http://localhost:8080/\" target=\"_blank\">Powrót do strony głównej</a><br/>";
+    }
+
+    @ResponseBody
+    @GetMapping("/list")
+    public String getProductsPerCategory(@RequestParam String category){
+        List<Product> productList = productRepository.findPerCategory(Category.valueOf(category));
```

Comment:
Popatrz teraz na metodę `getAll`. One się nie różnią między sobą - jedyne co je różni to metoda szukająca produktów w repozytorium. Możesz więc to zrobić w jednej metodzie gdzie ustawisz paramter `category` na opcjonalny i sprawdzisz ifem czy został podany czy nie (parametr != null) i wykonasz odpowiednią metodę repozytorium.